packages/core/test/interpreter/race.js

```diff
@@ -73,33 +73,33 @@ test('saga race between array of effects handling', assert => {
 test('saga race between effects: handle END', assert => {
   assert.plan(1)
 
-  let actual = []
+  let cancelled = false
   const timeout = deferred()
 
   const middleware = sagaMiddleware()
   const store = applyMiddleware(middleware)(createStore)(() => {})
 
   function* genFn() {
-    actual.push(
+    try {
       yield io.race({
         event: io.take('action'),
         timeout: timeout.promise,
-      }),
-    )
+      })
+    } finally {
+      cancelled = true
```

Comment:
It seems incorrect to do assignment in finally block. `cancelled = true` will always executes here regardless of the race result.

Reply:
oops. forgot to remove delay.

Reply:
I aligned the test with happy path version to  emphasise the difference between them
https://github.com/redux-saga/redux-saga/blob/af4b1b41f6f336bb4668fe2be569819a5ca3ee62/packages/core/test/interpreter/race.js#L8